### PR TITLE
feat: implement nLockTime finality check in transaction processing

### LINKED ISSUE #490: [FEATURE] check nLockTimeIsFinal during the process action
The task is to address the TODO here: https://github.com/bsv-blockchain/go-wallet-toolbox/blob/5f3738f490fc72fb9d318a42e6dfa852b249309c/pkg/storage/internal/actions/process.go#L106

Equivalent in TS: https://github.com/4chain-ag/wallet-toolbox/blob/6d63e74e57dfa65e04bc6cef8683c3c1e8330d08/src/storage/methods/processAction.ts#L288-L292

### PR DESCRIPTION
**Pull request by jakubmkowalski**
Closes: #490
This pull request introduces comprehensive validation and testing for transaction `nLockTime` finality in the transaction processing logic. It ensures that transactions are only processed if their `nLockTime` is considered final, improving the correctness and security of transaction handling. The most important changes are grouped below:

**Transaction finality enforcement:**

* Added a call to the new `NLockTimeIsFinal` service method in the `processNewTx` function to check transaction finality before processing. If the check fails or the transaction is not final, an error is returned.
* Extended the `Services` interface to include the new `NLockTimeIsFinal` method, which determines if a transaction or locktime is final.

**Testing for nLockTime finality:**

* Added extensive tests in `provider_process_action_test.go` to verify correct handling of various `nLockTime` scenarios, including successful finality, failure cases, service errors, and threshold boundary conditions. These tests ensure that the new logic is robust and correctly integrated.
* Updated test imports to include necessary dependencies for the new tests.## Description of Changes

### CODE REVIEW

File: pkg/storage/provider_process_action_test.go [hunk 1]
```diff
@@ -449,3 +451,252 @@ func TestProcessActionARCReturnNoBody(t *testing.T) {
...
+	given, cleanup := testabilities.Given(t)
+	defer cleanup()
+
+	activeStorage := given.Provider().
+		WithRandomizer(randomizer.NewTestRandomizer()).
+		GORM()
+
+	createActionResult, originalTx := given.Action(activeStorage).Created()
+
+	const nLockTimeThreshold = uint32(500_000_000)
```
[the remark this answers is not in the document]
Comment (jakubmkowalski) [reply pkg/storage/provider_process_action_test.go]: constant added

**Review by dorzepowski — APPROVED** [review]

**Review by mgosek-4chain — APPROVED** [review]

**Review by ac4ch — APPROVED** [review]
